1.86 cherrypicks

Duciwuci opened the pull request:
Well, this has a breakings: https://yake.cloud/release-notes/v1.86/

But the breakings needs to be fixed on our side. Therefore here are just the Cherries :)

timebertt committed:
Use external address in gardenlet bootstrap kubeconfig

timebertt committed:
Switch `garden-kube-apiserver` helm chart to `23ke` `GitRepository`

Kumm-Kai committed:
garden-kube-apiserver make featureGates configurable (#4)

nschad committed:
bump helmchart garden-kube-apiserver (#5)

Signed-off-by: Niclas Schad <niclas.schad@stackit.de>

Duciwuci commented:
Part of https://github.com/stackitcloud/ske-base/issues/917

nschad commented on helmcharts/garden-kube-apiserver/templates/deployment-kube-apiserver.yaml:
```diff
@@ -122,6 +122,13 @@ spec:
         - --service-account-issuer=https://kubernetes.default.svc
         - --service-account-signing-key-file=/srv/kubernetes/service-account-key/service_account.key
         - --service-account-key-file=/srv/kubernetes/service-account-key/service_account.key
+        {{- if .Values.apiServer.featureGates }}
```
Can we check if this is not possible now via the upstream chart?

Duciwuci replied:
Si Senor, lemme check

Duciwuci replied:
Well I don't see anything which makes feature gates configurable. Not even in latest: https://github.com/YAKEcloud/yake/blob/release-v1.93/helmcharts/garden-kube-apiserver/templates/deployment-kube-apiserver.yaml

Duciwuci replied:
Or am I blind?

nschad replied:
correct! maybe we should contribute it

Duciwuci commented on gardener/kube-apiserver.yaml:
```diff
@@ -18,11 +18,10 @@ spec:
-        kind: HelmRepository
-        name: gardener-community-charts
+        kind: GitRepository
+        name: 23ke
```
Shouldn't we move this permanently to yake?

Duciwuci replied:
https://yake.cloud/release-notes/v1.86/

Yes we should

Duciwuci commented on gardener/kube-apiserver.yaml:
```diff
@@ -18,11 +18,10 @@ spec:
-        kind: HelmRepository
-        name: gardener-community-charts
+        kind: GitRepository
+        name: yake
```
@timebertt I moved `23ke` to `yake`. 

According to new https://yake.cloud/release-notes/v1.86/

MichaelEischer commented:
/lgtm

looks like it could work. But please test run it before merging

Duciwuci committed:
Downgrade etcd Chart to 5.3.2

The upgrade to 6.0.0 includes breaking changes which restarts etcd
and requires an announcement.

MichaelEischer reviewed (approved):
Let's keep to downgraded etcd chart for now.
